src/util/logger-utils.ts

```diff
@@ -0,0 +1,73 @@
+import { Writable } from "stream";
+import safeJsonStringify from "safe-json-stringify";
+import bformat from "bunyan-format";
+import { createHashWithSharedSecret } from "utils/encryption";
+
+const SENSITIVE_DATA_FIELDS = ["jiraHost", "orgName", "repoName", "userGroup", "userGroup", "aaid", "username"];
+// For any Micros env we want the logs to be in JSON format.
+// Otherwise, if local development, we want human readable logs.
+const outputMode = process.env.MICROS_ENV ? "json" : "short";
+
+//TODO Remove this code when there will be convenient way to do it in Probot.
+//  See https://github.com/probot/probot/issues/1577
+export const filterHttpRequests = (record: Record<string, any>, filteredLoggerName: string) => {
+	const { msg, name } = record;
+	if (name !== filteredLoggerName) {
+		return false;
+	}
+	return !!msg.match(/(GET|POST|DELETE|PUT|PATCH)/);
+};
+
+export class RawLogStream extends Writable {
+	private readonly  filteredHttpLoggerName: string;
+	private readonly unsafeStream: boolean;
+	private writeStream: NodeJS.WritableStream;
+
+	public constructor(filteredHttpLoggerName: string, unsafeStream = false) {
+		super({ objectMode: true });
+		this.filteredHttpLoggerName = filteredHttpLoggerName;
+		this.unsafeStream = unsafeStream;
+		this.writeStream = bformat({ outputMode, levelInString: true });
+	}
+
+	public async _write(record: any, encoding: BufferEncoding, next): Promise<void> {
+		const featureFlags = await import("config/feature-flags");
+		const { booleanFlag, BooleanFlags } = featureFlags;
+
+		// Skip unwanted logs
+		if (filterHttpRequests(record, this.filteredHttpLoggerName)) {
+			return next();
+		}
+
+		// JiraHost and active feature flag required for unsafe stream
+		if (this.unsafeStream && (!record.jiraHost || !await booleanFlag(BooleanFlags.LOG_UNSAFE_DATA, false, record.jiraHost))) {
+			return next();
+		}
+
+		const recordClone = { ...record };
+
+		// hash sensitive data or tag it unsafe
+		if (this.unsafeStream) {
+			this.tagUnsafeData(recordClone);
+		} else {
+			this.hashSensitiveData(recordClone);
+		}
+
+		const chunk = safeJsonStringify(recordClone) + "\n";
+		this.writeStream.write(chunk, encoding);
```

Comment:
Originally had
```
this.writeStream.write(chunk, encoding, (err) => {
    console.error(err)
});
```

But.... we override the console logging with our logger, so an error on error would keep calling itself. Apparently infinite loops are frowned upon🤷‍♂️

Reply:
But it will means `next()` will be called before callback is triggered since the `writeStream.write` return immediately.

How about 
```
this.writeStream.write(chunk, encoding, (err) => { 
   if(err) { 
     //donothing, just next
     next();
   } else {
      next();
   }
});
```

Reply:
above is like 
```
this.writeStream.write(chunk, encoding, next);
```

but more explicit